docs/contributing.rst

```diff
@@ -80,6 +80,45 @@ guidelines when you're going to report a new bug.
 Writing patches
 ===============
 
+Before you can submit a change to the Scrapy code base, you need to:
+
+#. Learn the basics of Git_ and Github_
+
+#. Fork_ the Scrapy repository
+
+#. Clone_ your fork
+
+#. Configure_ upstream repository
+
+#. Install Scrapy using::
+  
+         pip install -e .
+
+   NOTE: Consider setting up a `Virtual Environment`_ before installing Scrapy. Virtual environment is used to create an isolated environment for 
+   python projects
+
+.. _Clone: https://help.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository
```

Comment:
Should these definitions be at the end of the file, or at least, at the end of the section? For instance, `add_` is used after it's defined.

Reply:
@elacuesta ,you're right. 
The definitions should be grouped together at the end of the section, as it will be easier to change definitions later, and it will look more organised.